static/app/views/insights/sessions/views/overview.tsx

```diff
@@ -93,37 +61,100 @@ export function SessionsOverview() {
                 <ModulesOnboardingPanel moduleName={ModuleName.SESSIONS} />
               </ModuleLayout.Full>
             ) : (
-              <Fragment>
-                {view === MOBILE_LANDING_SUB_PATH && (
-                  <Fragment>
-                    {SESSION_HEALTH_CHARTS}
-                    <ModuleLayout.Half>
-                      <ReleaseSessionCountChart />
-                    </ModuleLayout.Half>
-                    <ModuleLayout.Half>
-                      <ReleaseSessionPercentageChart />
-                    </ModuleLayout.Half>
-                    <ModuleLayout.Full>
-                      <FilterWrapper>
-                        <FilterReleaseDropdown
-                          filters={filters}
-                          setFilters={setFilters}
-                        />
-                      </FilterWrapper>
-                      <ReleaseHealth filters={filters} />
-                    </ModuleLayout.Full>
-                  </Fragment>
-                )}
-                {view === FRONTEND_LANDING_SUB_PATH && SESSION_HEALTH_CHARTS}
-              </Fragment>
+              <ViewSpecificCharts view={view} filters={filters} setFilters={setFilters} />
             )}
           </ModuleLayout.Layout>
         </Layout.Main>
       </Layout.Body>
-    </React.Fragment>
+    </Fragment>
   );
 }
 
+function ViewSpecificHeader({view}: {view: DomainView | ''}) {
+  switch (view) {
+    case FRONTEND_LANDING_SUB_PATH:
+      return <FrontendHeader module={ModuleName.SESSIONS} />;
+    case MOBILE_LANDING_SUB_PATH:
+      return <MobileHeader module={ModuleName.SESSIONS} />;
+    default:
+      return null;
+  }
+}
+
+function ViewSpecificCharts({
+  view,
+  filters,
+  setFilters,
+}: {
+  filters: string[];
+  setFilters: (filter: string[]) => void;
+  view: DomainView | '';
+}) {
+  switch (view) {
```

Comment:
it's a longer function to have this switch inside of it.. but i hope that it's worth the tradeoff of being easier to read.